--- a/rtl/interfaces/axi_support/rv_iopmp_data_abstractor_axi.sv
+++ b/rtl/interfaces/axi_support/rv_iopmp_data_abstractor_axi.sv
@@ -126,7 +126,7 @@ assign axi_aux_req.aw.user      = slv_req_i.aw.user;
 
 // W
 assign axi_aux_req.w            = slv_req_i.w;
-assign axi_aux_req.w_valid      = slv_req_i.w_valid;
+assign axi_aux_req.w_valid      = (state_q == AXI_HANDSHAKE) ? slv_req_i.w_valid : 1'b0;
 
 // B
 assign axi_aux_req.b_ready      = slv_req_i.b_ready;
